### ISSUE TITLE
Running cwe_checker on arm64 and x86_64 (failed)

### ISSUE THREAD

**Issue by StepanGavrilov** [fkie-cad/cwe_checker#351]
Tried to run cwe_checker on arm64 (mac m1) and x86_64 (failed). I build cwe_checker from source.

On arm64 everything is great. For example
```
root:/src/cwe/tmp_files# ls -l
-rw-r--r-- 1 root root 7499 Sep 19 10:15 crackme0x01
root:/src/cwe/tmp_files# whereis cwe_checker
\cwe_checker: /root/.cargo/bin/cwe_checker
root:/src/cwe/tmp_files# cwe_checker crackme0x01 
[CWE676] (0.1) (Use of Potentially Dangerous Function) main (08048426) -> scanf
```

On x86_64 for example, for same file

```
cwe_checker crackme0x01 --quiet
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: ValueUnrepresentable { value: ApInt { len: BitWidth(96), digits: [Digit(8589934593), Digit(1548712)] }, destination_ty: U64 }, message: "Encountered a value (ApInt { len: BitWidth(96), digits: [Digit(8589934593), Digit(1548712)] }) that is unrepresentable by the destination type U64.", annotation: None }', src/cwe_checker_lib/src/intermediate_representation/runtime_memory_image.rs:134:44
stack backtrace:
   0: rust_begin_unwind
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/panicking.rs:142:14
   2: core::result::unwrap_failed
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/result.rs:1805:5
   3: cwe_checker_lib::intermediate_representation::runtime_memory_image::RuntimeMemoryImage::read
   4: cwe_checker_lib::analysis::function_signature::state::State::load_value
   5: <cwe_checker_lib::analysis::function_signature::context::Context as cwe_checker_lib::analysis::forward_interprocedural_fixpoint::Context>::update_def
   6: cwe_checker_lib::analysis::fixpoint::Computation<T>::compute_with_max_steps
   7: cwe_checker_lib::analysis::function_signature::compute_function_signatures
   8: cwe_checker_lib::AnalysisResults::compute_function_signatures
   9: cwe_checker::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.`
```

**Comment by Enkelmann** [comment]
If your `crackme0x01` binary is not from an exotic architecture with address lengths greater than 64 bits then this is a bug in the cwe_checker. Why this only fails for x86_64 is unclear to me. Maybe you have different versions of Ghidra installed on your two systems. Anyway, this should be irrelevant for the cwe_checker bug.

Can you share the affected `crackme0x01` binary? Else it will be difficult for me to pinpoint the root cause for the bug. Also the used Ghidra version might be relevant if the bug is only triggered with a specific Ghidra version.

**Comment by Enkelmann** [comment]
Thanks for sharing the example binary! As a quick update: I found the root cause of the bug and I am currently working on a fix. The fix should be ready next week.

**Comment by StepanGavrilov** [comment]
@Enkelmann thanks a lot

**Comment by Enkelmann** [comment]
The fix is now merged (see PR #353). And I confirmed that the cwe_checker does not crash anymore when analyzing your example binary.
